### ISSUE TITLE
How to connect?

### ISSUE THREAD

**Issue by Bobselby1** [tesLAXApp/tesLAX#34]
Looking all over the place and for the life of me I cannot find any instructions or options to connect the app to my MX+ OBD reader. What am I missing?

**Comment by jake-b** [comment]
Please see the help document in the app. (Tap upper left menu > help) and find “connecting to accessories”

The MX+ needs to be BOTH paired and connected in the Apple iOS settings app.  It should then appear in the app under “Accessories” > “Bluetooth”

**Comment by M481248** [comment]
Having similar problems, I have tried both a OBDLinkMX+ and S3XY commander (separately).  I paired MX+ via Bluetooth and S3XY via Wifi.  On TesLAX, I can see the MX+ in accessories in the TesLAX app, but it never connects.  See Photo below. For S3XY, i have tried Panda and Wifi with same result.  Any suggestions?


![WIN_20231023_12_52_38_Pro](https://github.com/tesLAXApp/tesLAX/assets/148797914/1b266caa-4783-43c1-acd1-c421201e4879)

**Comment by jake-b** [comment]
Hmm.  That's two separate parts of the app that aren't working for you.  Are you getting data from other apps?  Wondering if it's a wiring issue.  What wiring harness are you using?

Wifi wise, you need to make sure you've allowed local network access, and I believe commander uses Panda, not ELM327-over-Wifi

Pop on discord to discuss, if you like: https://discord.gg/u2DkgvXG

**Comment by M481248** [comment]
Thanks for that, I will play around a bit more before connecting on discord.  FYI - just tried on my iphone 13 running IOS 16 and it works,  the Ipad (IPadOS 17) is where I have trouble.  Would like the iPad to be main device.
![WIN_20231023_13_52_46_Pro (2)](https://github.com/tesLAXApp/tesLAX/assets/148797914/9b6345fd-a796-4ad2-8c94-2b7966fc6d6e)

**Comment by Bobselby1** [comment]
Yes, the problem was the OBD2 adapter harness itself. I tried 3, the first two would not work. Ultimately got the one from EVOffer and it worked perfectly. Apparently the pin outs on some of the other ones just aren’t correct.

Sent from my iPhone

On Oct 23, 2023, at 12:55 PM, M481248 ***@***.***> wrote:

﻿

Having similar problems, I have tried both a OBDLinkMX+ and S3XY commander (separately). I paired MX+ via Bluetooth and S3XY via Wifi. On TesLAX, I can see the MX+ in accessories in the TesLAX app, but it never connects. See Photo below. For S3XY, i have tried Panda and Wifi with same result. Any suggestions?

[WIN_20231023_12_52_38_Pro]<https://user-images.githubusercontent.com/148797914/277413103-1b266caa-4783-43c1-acd1-c421201e4879.jpg>

**Comment by M481248** [comment]
My harness is good.  Scan my Tesla (android) works fine and it works with my iphone (iOS 16).  I believe it is an issue now with IpadOS17 running on the iPad.  Any idea if iPadOS17 is supported with Tes LAX?

**Comment by M481248** [comment]
Jake,Can confirm my issue is with the iPadOS 17 as I borrowed an iPad running is iPadOS 15 and was able to collect the data as expected. Have you heard of other issues with those running OS17 (assuming iOS 17 would give similar issues as iPadOS 17). ThanksOn Oct 23, 2023, at 13:11, jake-b ***@***.***> wrote:﻿
Hmm.  That's two separate parts of the app that aren't working for you.  Are you getting data from other apps?  Wondering if it's a wiring issue.  What wiring harness are you using?
Wifi wise, you need to make sure you've allowed local network access, and I believe commander uses Panda, not ELM327-over-Wifi
Pop on discord to discuss, if you like: https://discord.gg/u2DkgvXG

**Comment by jake-b** [comment]
Hi.  This is the first report of an issue specific to iPadOS.  I will check it out on my iPad mini and see if I can reproduce/debug.  Not sure why there would be a difference in behavior

**Comment by stromi-tesla-model3** [comment]
Can report there are no problems with Tes•LAX 2.4.2 (174) on iPadOS 17.0.3 on iPad mini (5th Gen).
A suggestion is to use "Forget this device" on any other device that previously connected to the OBD MX+.
Sometimes MX+ will switch devices.

**Comment by M481248** [comment]
Good to know Stomi - Thanks!   at least I now know its the device :)  I will play around with it.

**Comment by stromi-tesla-model3** [comment]
> Sometimes MX+ will switch devices.

Don't forget to apply any firmware updates to the MX+ via the OBDLink app. OBDLink may seem broken at startup, as it doesnt "connect to car", but with OBDLink and MX+ connected, you can check for and apply firmware updates.

**Comment by SPlatten** [comment]
What do you need to connect the app to a Tesla?  I have various ODBCII connectors and I've been investigating the CAN-BUS with SavvyCAN and connecting to my Tesla CAN-BUS via the centre console, but how would I connect Tes-LAX to the car?
